### ISSUE TITLE
Request for Information on Data Theorem Finding

### ISSUE THREAD

**Issue by scottluxenberg** [firebase/firebase-ios-sdk#12187]
### Description

This is bubbling back up https://github.com/firebase/firebase-ios-sdk/issues/5447

We have a finding in Firebase for `App Embeds SQL Query with Dynamic Input References:`

The issue linked above has a statement from Firebase saying 
`the %@ is only for hard coded strings (table names, column names) so that we can reuse the code internally. All user inputs are validated, sanitized and bound appropriately for each query. We have been through security audit of the source code before releasing.`

We brought this issue up to our security team, but they have said the comment is not enough attestation. Is there any releasable details of said security audit that shows this attack vector is indeed secure, that I can bring back to my security team? 

### Firebase SDK Version

10.11

### Xcode Version

4.2

### Installation Method

CocoaPods

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by scottluxenberg** [comment]
@paulb777 , yes the finding is actually in Analytics, my bad (I just put Crashlytics as that's our primary usage).

And I think if nothing has changed that's fine, I just want to know if there is a more detailed report to show that this is indeed not an issue and not jsut a GitHub comment - that's what our security department has asked for. Otherwise I have to file security exceptions - so just wondering if there is any releasble proof to show that the finding is indeed not an issue

**Comment by axle-os** [comment]
Hello! This is Phillip from the Data Theorem side. If it helps, would you please be able to provide some kind of symbol name (or w/e) for the sanitisation mentioned in #5447? Then, I can validate on my end from the disassembled code and provide an 'official' assurance from Data Theorem that we've confirmed that the SDK behaviour is fine. Would this work for your security team @scottluxenberg?

**Comment by codyd51** [comment]
Hello! This is Phillip from the Data Theorem side. If it helps, would you please be able to provide some kind of symbol name (or w/e) for the sanitisation mentioned in #5447? Then, I can validate on my end from the disassembled code and provide an 'official' assurance from Data Theorem that we've confirmed that the SDK behaviour is fine. Would this work for your security team @scottluxenberg?

**Comment by scottluxenberg** [comment]
Yeah, if you all validate that, an actual validation from Data Theorem would be enough to satisfy us

**Comment by scottluxenberg** [comment]
Any update on if this info is providable?

**Comment by paulb777** [comment]
Closing for staleness. Please comment or create a new issue referring to this one for reconsideration.
